src/Our.Umbraco.Ditto/Extensions/PublishedContentExtensions.cs

```diff
@@ -315,54 +315,33 @@ public static class PublishedContentExtensions
-            PropertyInfo propertyInfo)
+        /// <param name="instance">The instance to assign the value to.</param>
+        private static object GetRawValue(
+           IPublishedContent content,
+            Type type,
+            CultureInfo culture,
+            PropertyInfo propertyInfo,
+            ref object instance)
         {
-            var contentType = content.GetType();
-            var umbracoPropertyName = propertyInfo.Name;
-            var altUmbracoPropertyName = string.Empty;
-            var recursive = false;
-            object defaultValue = null;
-
-            var umbracoPropertyAttr = propertyInfo.GetCustomAttribute<UmbracoPropertyAttribute>();
-            if (umbracoPropertyAttr != null)
-            {
-                umbracoPropertyName = umbracoPropertyAttr.PropertyName;
-                altUmbracoPropertyName = umbracoPropertyAttr.AltPropertyName;
-                recursive = umbracoPropertyAttr.Recursive;
-                defaultValue = umbracoPropertyAttr.DefaultValue;
-            }
-
-            // Try fetching the value.
-            var contentProperty = contentType.GetProperty(umbracoPropertyName);
-            object propertyValue = contentProperty != null
-                                    ? contentProperty.GetValue(content, null)
-                                    : content.GetPropertyValue(umbracoPropertyName, recursive);
-
-            // Try fetching the alt value.
-            if ((propertyValue == null || propertyValue.ToString().IsNullOrWhiteSpace())
-                && !string.IsNullOrWhiteSpace(altUmbracoPropertyName))
-            {
-                contentProperty = contentType.GetProperty(altUmbracoPropertyName);
-                propertyValue = contentProperty != null
-                                    ? contentProperty.GetValue(content, null)
-                                    : content.GetPropertyValue(altUmbracoPropertyName, recursive);
-            }
-
-            // Try setting the default value.
-            if ((propertyValue == null || propertyValue.ToString().IsNullOrWhiteSpace())
-                && defaultValue != null)
-            {
-                propertyValue = defaultValue;
-            }
-
-            return propertyValue;
+            // Check the property for an associated value attribute, otherwise fall-back on expected behaviour.
+            var valueAttr = propertyInfo.GetCustomAttribute<DittoValueResolverAttribute>(true)
+                ?? new UmbracoPropertyAttribute();
+
+            //TODO: Only create one context and share between GetRawValue and SetTypedValue?
+            var descriptor = TypeDescriptor.GetProperties(instance)[propertyInfo.Name];
+            var context = new PublishedContentContext(content, descriptor);
+
+            // Get the value from the custom attribute.
+            //TODO: Cache these?
+            var resolver = (DittoValueResolver)Activator.CreateInstance(valueAttr.ResolverType);
```

Comment:
A wee heads up .We're using a single place for creating objects. An extension `type.GetInstance()` It's so we can keep the api consistent for creation as the overloads of that extension use much faster initialisation code than the default.